src/mcts/search.cc

```diff
@@ -1214,9 +1226,24 @@ void SearchWorker::ExtendNode(Node* node) {
       return;
     }
 
-    if (history_.Last().GetRepetitions() >= 2) {
+    // Mark two-fold repetitions as draws according to settings
+    const auto repetitions = history_.Last().GetRepetitions();
+    const auto twofolddrawlevel = history_.Last().GetTwoFoldDrawLevel();
+    if (repetitions >= 2) {
       node->MakeTerminal(GameResult::DRAW);
       return;
+    } else if (repetitions == 1 && twofolddrawlevel > 0) {
+      if (twofolddrawlevel == 3) {
+        // always mark as draw
+        node->MakeTerminal(GameResult::DRAW, 0.0f, Node::Terminal::TwoFold);
+      } else if (twofolddrawlevel == 2) {
+        // only mark as draw if depth >= 4
+        node->MakeTerminal(GameResult::DRAW, 0.0f, Node::Terminal::TwoFold);
+      } else if (twofolddrawlevel == 1) {
+        // check whether first repetition happened at root or in the tree
```

Comment:
This function doesn't exist yet, but should be a simple copy+paste from `position.cc:GetRepetitions()`

Reply:
It does exist now.